DRODLib/Character.cpp

```diff
@@ -5093,6 +5214,57 @@ void CCharacter::SetVariable(const CCharacterCommand& command, CCurrentGame* pGa
 	}
 }
 
+//*****************************************************************************
+void CCharacter::SetArrayVariable(
+	const CCharacterCommand& command,
+	CCurrentGame* pGame,
+	CCueEvents& CueEvents)
+{
+	const UINT varIndex = command.w;
+	if (pGame->pHold && !pGame->pHold->IsArrayVar(varIndex))
```

Comment:
Hmm...I agree that we need to guard against messing with vars that are non-arrays, but where is the logic that turns a non-initialized var into an array in the first place? Wouldn't that be done here?

Reply:
The current way things are wired up, script vars being array or non-array is sort of like static typing, based on whether the variable name starts with an array signifier (`#` or `@`). In the same way `CCharacter::SetVariable` checks if a var is local using the pre-built map in `CDbHold`, the check for a var being an array is done using a similar map. This gets constructed when the hold is loaded in, so an array var is effectively always an array, even without being initialized. (the check against changing the type via renaming doesn't use this map because it's not always up to date when editing, but it follows the same principle)